tests/acceptance/ghost-desktop-test.js

```diff
@@ -10,23 +10,27 @@ import startApp from '../helpers/start-app';
 import destroyApp from '../helpers/destroy-app';
 import { authenticateSession } from 'ghost-admin/tests/helpers/ember-simple-auth';
 
-const originalUserAgent = window.navigator.userAgent;
-
-const _setUserAgent = function (userAgent) {
-    let userAgentProp = {get() { return userAgent; }, configurable: true};
-    Object.defineProperty(window.navigator, 'userAgent', userAgentProp);
-};
-
-const stubUserAgent = function (userAgent) {
-    if (window.navigator.userAgent !== userAgent) {
-        _setUserAgent(userAgent);
+const originalAgent = window.navigator.userAgent;
+
+const setUserAgent = function (userAgent) {
+    let userAgentProp = {
+        get() {
+            return userAgent;
+        },
+        configurable: true
+    };
+
+    try {
```

Comment:
Firefox and Chrome handle navigation overrides differently than Safari. Safari doesn't let you change the `userAgent` property directly, so the `catch` block handles the exception from the first method and then creates a dummy `window.navigator` object that works in Safari.